Tools/CISupport/ews-build/steps.py

```diff
@@ -4905,25 +4977,30 @@ def evaluateCommand(self, cmd):
         self.setProperty('build_summary', commit_summary)
         self.addURL(self.identifier, self.url_for_identifier(self.identifier))
 
-        return rc
+        defer.returnValue(rc)
 
     def url_for_hash_details(self, hash):
         return '{}{}/json'.format(COMMITS_INFO_URL, hash)
 
     def url_for_identifier(self, identifier):
         return '{}{}'.format(COMMITS_INFO_URL, identifier)
 
+    @defer.inlineCallbacks
     def identifier_for_hash(self, hash):
         try:
-            response = requests.get(self.url_for_hash_details(hash), timeout=60)
-            if response and response.status_code == 200:
-                return response.json().get('identifier', '{}'.format(hash)).replace('@trunk', '@main')
+            response = yield TwistedAdditions.request(
+                url=f'{COMMITS_INFO_URL}{hash}/json', logger=print,
+            )
+            if response.status_code == 200:
```

Comment:
can response be None? if so we `response.status_code` will raise an exception. We should safeguard against that.